statsmodels/discrete/count_model.py

```diff
@@ -200,6 +200,11 @@ def fit_regularized(self, start_params=None, method='l1',
             alpha=0, trim_mode='auto', auto_trim_tol=0.01, size_trim_tol=1e-4,
             qc_tol=0.03, **kwargs):
 
+        if method not in ['l1', 'l1_cvxopt_cp']:
+            # FIXME: This should be a ValueError right?
+            raise TypeError("argument method == %s, which is not handled" %
```

Comment:
Please no old style format strings.  Also, needs test coverage since isn't hit.  Same for other.

In case you don't have the codecov plugin for Chrome -- you see ticks like this on github.

![image](https://user-images.githubusercontent.com/5585221/32023781-40e971b2-b9d2-11e7-8d71-7d2fcc13f86b.png)